lib/jekyll/post.rb

```diff
@@ -307,7 +315,7 @@ def extract_excerpt
     end
 
     def generate_excerpt?
-      !(site.config['excerpt_separator'].to_s.empty?)
+      !site.config['excerpt_separator'].to_s.empty? || !excerpt_separator.empty?
```

Comment:
If we move the fallback into `Post::excerpt_separator` (see above comments), checking `site.config` won't be necessary here.